### PR TITLE
Bsric0

### PR DESCRIPTION
**Pull request by jsandham**
Adding bsric02

### CODE REVIEW

File: library/src/hcc_detail/hipsparse.cpp [hunk 1]
```diff
@@ -5490,6 +5500,488 @@ hipsparseStatus_t hipsparseZcsrilu02(hipsparseHandle_t         handle,
...
+
+
+
+
+
+
+
+
+
+
```
Comment (ntrost57) [inline library/src/hcc_detail/hipsparse.cpp:5513]: Can you remove the empty lines?

File: library/src/hcc_detail/hipsparse.cpp [hunk 2]
```diff
@@ -5490,6 +5500,488 @@ hipsparseStatus_t hipsparseZcsrilu02(hipsparseHandle_t         handle,
...
+hipsparseStatus_t
+    hipsparseXbsric02_zeroPivot(hipsparseHandle_t handle, bsric02Info_t info, int* position)
+{
+    // Obtain stream, to explicitly sync (cusparse bsric02_zeropivot is blocking)
```
Comment (ntrost57) [inline library/src/hcc_detail/hipsparse.cpp:5559]: Did you verify that cusparse is actually blocking?

File: library/src/hcc_detail/hipsparse.cpp [hunk 3]
```diff
@@ -5490,6 +5500,488 @@ hipsparseStatus_t hipsparseZcsrilu02(hipsparseHandle_t         handle,
...
+                                             const hipsparseMatDescr_t descrA,
+                                             const float*              bsrValA,
+                                             const int*                bsrRowPtrA,
+                                             const int*                bsrColIndA,
+                                             int                       blockDim,
+                                             bsric02Info_t             info,
+                                             hipsparseSolvePolicy_t    policy,
+                                             void*                     pBuffer)
+{
+    // Obtain stream, to explicitly sync (cusparse bsric02_analysis is blocking)
```
Comment (ntrost57) [inline library/src/hcc_detail/hipsparse.cpp:5734]: Did you verify cusparse is blocking?

File: library/src/hipsparse_module.f90 [hunk 4]
```diff
@@ -3153,6 +3170,256 @@ function hipsparseZcsrilu02(handle, m, nnz, descr, csrVal, csrRowPtr, &
...
+                result(c_int) &
+                bind(c, name = 'hipsparseZbsric02_bufferSize')
+            use iso_c_binding
+            implicit none
+            type(c_ptr), value :: handle
+            integer(c_int), value :: dir
+            integer(c_int), value :: mb
+            integer(c_int), value :: nnzb
+            type(c_ptr), intent(in), value :: descr
+            type(c_ptr), intent(in), value :: bsrVal
```
Comment (ntrost57) [inline library/src/hipsparse_module.f90:3253]: bsrVal is non-const (also in other precisions)

File: library/src/hcc_detail/hipsparse.cpp (hunk 2 above)
Comment (jsandham) [reply library/src/hcc_detail/hipsparse.cpp:5559]: Their documentation says cusparseXbsric0_zeroPivot is blocking as they call cudaDeviceSynchronize but I did not verify it.

File: library/src/hcc_detail/hipsparse.cpp (hunk 3 above)
Comment (jsandham) [reply library/src/hcc_detail/hipsparse.cpp:5734]: Their documentation says it does not support asynchronous execution but I did not verify it. I am actually not sure how I would verify it

File: library/src/hipsparse_module.f90 (hunk 4 above)
Comment (jsandham) [reply library/src/hipsparse_module.f90:3253]: Are you referring to the fact that in cusparseSbsric0_bufferSize etc pass bsrVAl by non const? I think this is a bug as bsrVAl is not modified by this function. I made the rocsparse routines pass it const.

File: library/src/hcc_detail/hipsparse.cpp (hunk 1 above)
Comment (jsandham) [reply library/src/hcc_detail/hipsparse.cpp]: Fixed in latest push

File: library/src/hipsparse_module.f90 (hunk 4 above)
Comment (jsandham) [reply library/src/hipsparse_module.f90:3253]: Looks like we do the same for csric0 even though cusparse uses non-const

**Pull request merged**